qemu/tests/watchdog.py

```diff
@@ -282,6 +282,30 @@ def hotplug_unplug_watchdog_device():
                 test.fail("Oops, find watchdog pci, unplug failed")
             logging.info("The WDT remove successfully")
 
+    def stop_cont_test():
+        """
+        Check if the emulated watchdog devices work properly with the stop/
+        continue operation
+        """
+
+        response_timeout = int(params.get("response_timeout", '240'))
```

Comment:
This value should be got from guest by querying the attribute of the module. Or, you should reload the module with setting its attribute to the value provided this param.

Reply:
@luckyh Line40 and Line42 don't specify the heartbeat value, so it is the default value 30s, if so, do I still need to query the value of the module?

Reply:
you didn't restrict the test to be only run with i6300esb, as ib700 used another default value, I think it should be required.

Reply:
@luckyh I just checked the default value of ib700, it is 30s.
#modinfo ib700wdt
filename:       /lib/modules/3.10.0-907.el7.x86_64/kernel/drivers/watchdog/ib700wdt.ko.xz
alias:          char-major-10-130
license:        GPL
description:    IB700 SBC watchdog driver
author:         Charles Howes <chowes@vsol.net>
retpoline:      Y
rhelversion:    7.6
srcversion:     5B7E918EB930218625CCD6F
depends:        
intree:         Y
vermagic:       3.10.0-907.el7.x86_64 SMP mod_unload modversions 
signer:         Red Hat Enterprise Linux kernel signing key
sig_key:        2A:9F:B6:3C:58:E4:8A:34:C8:E9:AA:96:C8:69:E5:D6:80:CC:92:49
sig_hashalgo:   sha256
parm:           timeout:Watchdog timeout in seconds. 0<= timeout <=30, default=30. (int)
parm:           nowayout:Watchdog cannot be stopped once started (default=0) (bool)

Reply:
Or you mean diag288 has a different value? I will also check it.

Reply:
No timeout param for s390x. @qunfang is confirming with it.
# modinfo diag288_wdt
modinfo diag288_wdt
filename:       /lib/modules/4.14.0-113.el7a.s390x/kernel/drivers/watchdog/diag288_wdt.ko.xz
alias:          vmwatchdog
alias:          char-major-10-130
description:    System z diag288  Watchdog Timer
author:         Philipp Hachtmann <phacht@de.ibm.com>
author:         Arnd Bergmann <arndb@de.ibm.com>
license:        GPL
rhelversion:    7.6
srcversion:     99B138785F7C64C747C7170
depends:        
intree:         Y
name:           diag288_wdt
vermagic:       4.14.0-113.el7a.s390x SMP mod_unload modversions 
parm:           cmd:CP command that is run when the watchdog triggers (z/VM only) (string)
parm:           conceal:Enable the CONCEAL CP option while the watchdog is active (z/VM only) (bool)
parm:           nowayout:Watchdog cannot be stopped once started (default = CONFIG_WATCHDOG_NOWAYOUT) (bool)

Reply:
@luckyh Do you accept it or still request changes? Thanks.

Reply:
About diag288_wdt on s390x,  timeout value is 30s, it's hardcode in kernel:
	if (__diag288_lpar(WDT_FUNC_INIT, 30, LPARWDT_RESTART)) {
			pr_err("The watchdog cannot be initialized\n");
			return -EINVAL;
		}